### ISSUE TITLE
MySQL 8.0 issue with is_private

### ISSUE THREAD

**Issue by trajche** [deliciousbrains/wp-amazon-s3-and-cloudfront#531]
In the Pro version, when using MySQL 8.0 (Managed by DigitalOcean), there is an error when offloading media to S3 bucket (DigitalOcean Spaces):

```
WordPress database error Incorrect integer value: '' for column 'is_private' at row 1 for query INSERT INTO `wp_as3cf_items` (`bucket`, `extra_info`, `is_private`, `original_path`, `original_source_path`, `path`, `provider`, `region`, `source_id`, `source_path`, `source_type`) VALUES ('offload', 'a:1:{s:13:\"private_sizes\";a:0:{}}', '', '2020/03/71spssr_2ba-l._ul1500.jpg', '2020/03/71spssr_2ba-l._ul1500.jpg', '2020/03/71spssr_2ba-l._ul1500.jpg', 'do', 'fra1', 661820, '2020/03/71spssr_2ba-l._ul1500.jpg', 'media-library') made by DeliciousBrains\WP_Offload_Media\Items\Item::save
```

This is fixed by changing line 65 and checking if $is_private is empty, then setting it to a default of 0:

```
//Fixing MySQL 8.0 Bug by TJ
if(empty($is_private)) {
   $this->is_private  = 0;
} else {
   $this->is_private  = $is_private;
}
```

p.s. This works fine from localhost when using an older MySQL version.

Thanks for a great plugin.

**Comment by ianmjones** [comment]
Thanks, we'll look into this and will post here with any questions we may have.

**Comment by ianmjones** [comment]
This should be fixed in WP Offload Media (Lite) 2.4 as a side-effect of a fix for HyperDB that I believe stumbled in the same area.
